src/config/config.c

```diff
@@ -416,11 +419,14 @@ __config_next(WT_CONFIG *conf, WT_CONFIG_ITEM *key, WT_CONFIG_ITEM *value)
     if (conf->depth <= conf->top && key->len > 0)
         return (0);
 
-    /* We're either at the end of the string or we failed to parse. */
-    if (conf->depth == 0)
-        return (WT_NOTFOUND);
+    if (key->len == 0 && conf->go == gostring)
+        return (__config_err(conf, "Unbalanced quotes", EINVAL));
```

Comment:
This is the actual check for unbalanced string. We reached the end, but the state is still `string`, meaning that the opening `"` was never matched with the closing pair.